RULES/cc/cc.go

```diff
@@ -296,6 +299,19 @@ func (bin Binary) build(ctx core.Context) {
 		}
 	}
 
+	libsPre := []core.Path{}
+	for _, dep := range bin.DepsPre {
+		lib := dep.CcLibrary(toolchain)
+		ins = append(ins, lib.Out)
+		libsPre = append(libsPre, lib.Out)
+	}
+
+	for _, dep := range bin.DepsPost {
+		lib := dep.CcLibrary(toolchain)
+		ins = append(ins, lib.Out)
+		otherLibs = append(otherLibs, lib.Out)
```

Comment:
What happend if any of the `libsPre` or `libsPost` have `AlwaysLink` set to true?
Maybe it makes more sense to add the `-Wl,-whole-archive` flags on a per-lib basis? So have three slices with the pre , post and normal libs, combine that into one ordered list and then generate the command and prefix each lib with the proper flag depending on its `AlwaysLink` status. WDYT?

Reply:
Good idea, updated